virtcontainers/qemu.go

```diff
@@ -961,6 +961,21 @@ func (q *qemu) stopSandbox() error {
 
 	err := q.qmpSetup()
 	if err != nil {
+		// Ignore any "connection refused" error and assume that the Sandbox
```

Comment:
Would you mind putting this code in a separate function `isConnexionRefusedError(err)` to streamline the logic here and clarify the intent?

Reply:
@rhafer Thanks for raising this issue that you are seeing with newer containerd versions. 
Its not clear to me though why the sandbox has already gone at this stage given the behaviour your referenced in the link. Will need to take a closer look at this. 
We will probably need to have the fix in such a way that it applies to all supported hypervisors and not just qemu.

Reply:
> Would you mind putting this code in a separate function isConnexionRefusedError(err) to streamline the logic here and clarify the intent?

:+1:

> Its not clear to me though why the sandbox has already gone at this stage given the behaviour your referenced in the link. Will need to take a closer look at this.

The issue happens when the qemu process gets killed (or crashes). I used https://github.com/kata-containers/tests/blob/master/integration/stability/hypervisor_stability_kill_test.sh as a reproducer. (Basically what is described in #2611)

> We will probably need to have the fix in such a way that it applies to all supported hypervisors and not just qemu.

Hm, yeah. I haven't checked the other hypervisors. Not sure if their implementations suffer from the same issue. A way to fix this in a more general way might be always call `deleteSandbox` with  `force=true` here https://github.com/kata-containers/runtime/blob/master/cli/delete.go#L121. However I think this might be ignore a few real errors.

Reply:
/cc @likebreath.